commit 385d2ea937bf5a9ec3b40ff167141ff58eb4143d
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 16:10:04 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-port-environment-option.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
index a558cdb0a..480b680b0 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -2661,6 +2661,12 @@ proc action_version { action portlist opts } {
 }
 
 
+proc action_environment { action portlist opts } {
+    set status [macports::environment]
+    return $status
+}
+
+
 proc action_platform { action portlist opts } {
     global macports::os_platform macports::os_major macports::os_arch
     if {![macports::ui_isset ports_quiet]} {
@@ -4107,6 +4113,7 @@ set action_array [dict create \
     diagnose    [list action_diagnose       [ACTION_ARGS_NONE]] \
     \
     version     [list action_version        [ACTION_ARGS_NONE]] \
+    environment [list action_environment    [ACTION_ARGS_NONE]] \
     platform    [list action_platform       [ACTION_ARGS_NONE]] \
     \
     uninstall   [list action_uninstall      [ACTION_ARGS_PORTS]] \
@@ -4364,6 +4371,9 @@ proc parse_options { action ui_options_name global_options_name } {
             set opts [string range $arg 1 end]
             foreach c [split $opts {}] {
                 switch -- $c {
+                    e {
+                        set ui_options(ports_env) yes
+                    }
                     v {
                         set ui_options(ports_verbose) yes
                     }
